include/wlr/xwayland.h

```diff
@@ -53,6 +57,11 @@ struct wlr_xwayland_surface {
 	xcb_atom_t *protocols;
 	size_t protocols_len;
 
+	uint32_t motif_hints[5];
+	#ifdef HAS_XCB_ICCCM
+	xcb_size_hints_t size_hints;
```

Comment:
Not sure how to do this.

Reply:
You'll want to leave the data in the struct even if the library isn't in use. Figure out how big xcb_size_hints_t is and set aside the same amount of space.